### PR TITLE
Canbus & others : Add initial support for GEM with further re-factor to be done. 

### PR DESCRIPTION
**Pull request by Capri2014**

### CODE REVIEW

File: modules/canbus/vehicle/gem/protocol/brake_cmd_6b.cc [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/******************************************************************************
+ * Copyright 2017 The Apollo Authors. All Rights Reserved.
```
Comment (startcode) [inline modules/canbus/vehicle/gem/protocol/brake_cmd_6b.cc]: Change to 2018

File: modules/canbus/vehicle/gem/protocol/brake_motor_rpt_2_71.cc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+  Byte t1(bytes + 1);
+  int32_t t = t1.get_byte(0, 8);
+  x <<= 8;
+  x |= t;
+
+  x <<= 16;
+  x >>= 16;
+
+  int ret = x + -40.000000;
```
Comment (startcode) [inline modules/canbus/vehicle/gem/protocol/brake_motor_rpt_2_71.cc]: should this 40.0000 be an integer?

File: modules/canbus/vehicle/gem/protocol/brake_cmd_6b.cc (hunk 1 above)
Comment (Capri2014) [reply modules/canbus/vehicle/gem/protocol/brake_cmd_6b.cc]: done

File: modules/canbus/vehicle/gem/protocol/brake_motor_rpt_2_71.cc (hunk 2 above)
Comment (Capri2014) [reply modules/canbus/vehicle/gem/protocol/brake_motor_rpt_2_71.cc]: @startcode Done

**Pull request merged**